### ISSUE TITLE
Erreur dans UseMethod("compile") 

### ISSUE THREAD

**Issue by OusmaneS17** [rstudio/keras#1436]
Bonjour, j'ai construit un modèle en utilisant keras  mais j'ai eu une erreur de compilation comme celle-ci: 
model <- keras_model(inputs = input_layer, outputs = list(output_revenu, output_data, output_voix))
> summary(model)
<Functional name=functional_1, built=True>
> model <- compile(
+   model,
+   optimizer = optimizer_adam(),
+   loss = "mean_squared_error",
+   metrics = list("mean_squared_error"),
+   loss_weights = list(output_revenu = 1, output_data = 1, output_voix = 1)
+ )
Erreur dans UseMethod("compile") : 
  pas de méthode pour 'compile' applicable pour un objet de classe "c('keras.src.models.functional.Functional', 'keras.models.functional.Functional', 'keras.src.ops.function.Function', 'keras.ops.function.Function', 'keras.src.models.model.Model', 'keras.models.model.Model', 'keras.src.backend.tensorflow.trainer.TensorFlowTrainer', 'keras.backend.tensorflow.trainer.TensorFlowTrainer', 'keras.src.trainers.trainer.Trainer', 'keras.trainers.trainer.Trainer', 'keras.src.layers.layer.Layer', 'keras.layers.layer.Layer', 'keras.src.backend.tensorflow.layer.TFLayer', 'keras.backend.tensorflow.layer.TFLayer', 'keras.src.backend.tensorflow.trackable.KerasAutoTrackable', 'keras.backend.tensorflow.trackable.KerasAutoTrackable', 'tensorflow.python.trackable.autotrackable.AutoTrackable', 'tensorflow.python.trackable.base.Trackable', 'keras.src.ops.operation.Operation', 'keras.ops.operation.Operation', 'python.builtin.object')"

**Comment by t-kalinowski** [comment]
Hello, we are only able to provide support in English. Can you please update your post to English? Using something like https://translate.google.com might be helpful.

**Comment by OusmaneS17** [comment]
hello,

I built a model using keras but got a compilation error like this:
model <- keras_model(inputs = input_layer, outputs = list(output_revenue, output_data, output_voice))

summary (template)

> model <- compile( + model, + optimizer = optimizer_adam(), + loss = "mean_squared_error", + metrics = list("mean_squared_error"), + loss_weights = list(output_revenue = 1, output_data = 1, output_voice = 1) + ) Error in UseMethod("compile"): no method for 'compile' applicable for an object of class "c('keras.src.models.functional.Functional', 'keras.models.functional.Functional', ' keras.src.ops.function.Function', 'keras.ops.function.Function', 'keras.src.models.model.Model', 'keras.models.model.Model', 'keras.src.backend. tensorflow.trainer.TensorFlowTrainer', 'keras.backend.tensorflow.trainer.TensorFlowTrainer', 'keras.src.trainers.trainer.Trainer', 'keras.trainers.trainer.Trainer', 'keras.src.layers.layer. Layer', 'keras.layers.layer.Layer', 'keras.src.backend.tensorflow.layer.TFLayer', 'keras.backend.tensorflow.layer.TFLayer', 'keras.src.backend.tensorflow.trackable. KerasAutoTrackable', 'keras.backend.tensorflow.trackable.KerasAutoTrackable', 'tensorflow.python.trackable.autotrackable.AutoTrackable', 'tensorflow.python.trackable.base.Trackable', 'keras.src.ops.operation.Operation' , 'keras.ops.operation.Operation', 'python.builtin.object')"

**Comment by OusmaneS17** [comment]
"It's with Keras. Attached you will find the code and the database.
The version of my CRAN is:"

version$version.string
[1] "R version 4.4.0 (2024-04-24 ucrt)"


Ousmane SOW

Élève Analyste Statisticien à l’ENSAE de Dakar

*Tel :* +221 70-571-83-03

<https://www.linkedin.com/in/ousmane-sow-847b56226/>

**Comment by t-kalinowski** [comment]
Hello, I don't beleive github supports email replies with attachments. Can you please paste the output from running `reprex::reprex()` into the issue thread using the github UI: https://github.com/rstudio/keras/issues/1436

**Comment by OusmaneS17** [comment]
reprex::reprex()
#> ℹ Non-interactive session, setting `html_preview = FALSE`.
#> CLIPR_ALLOW has not been set, so clipr will not run interactively
#> Error in switch(where, expr = stringify_expression(x_expr), clipboard = ingest_clipboard(), : EXPR doit être un vecteur de longueur 1


Created on 2024-04-25 with [reprex v2.0.2](https://reprex.tidyverse.org/)

**Comment by t-kalinowski** [comment]
Closing for now since I can't reproduce the error.
